WalletWasabi/Extensions/NBitcoinExtensions.cs

```diff
@@ -263,5 +263,15 @@ public static void SortByAmount(this TxOutList list)
 		{
 			list.Sort((x, y) => x.Value.CompareTo(y.Value));
 		}
+
+		public static void SortByAmount(this TxInList list, List<Coin> coins)
+		{
+			var map = new Dictionary<TxIn, Coin>();
+			foreach (var coin in coins)
+			{
+				map.Add(list.Single(x => x.PrevOut == coin.Outpoint), coin);
+			}
+			list.Sort((x, y) => map.TryGet(x).Amount.CompareTo(map.TryGet(y).Amount));
```

Comment:
Please don't use `TryGet`, it is a NBitcoin extension method that returns `null` in case the key is not found, what is impossible in our case, so, it is not necessary because we know the item is there. Moreover, in case the item is not found we get a `NullReferenceException`. Use `[x]` instead.